noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -0,0 +1,309 @@
+mod types;
+
+// Minimal NFT implementation with `AuthWit` support that allows minting in public-only and transfers in both public
+// and private.
+contract NFT {
+    use dep::compressed_string::FieldCompressedString;
+    use dep::aztec::{
+        prelude::{NoteGetterOptions, NoteViewerOptions, Map, PublicMutable, SharedImmutable, PrivateSet, AztecAddress},
+        encrypted_logs::{encrypted_note_emission::encode_and_encrypt_note_with_keys},
+        hash::pedersen_hash, keys::getters::get_current_public_keys,
+        note::constants::MAX_NOTES_PER_PAGE, protocol_types::traits::is_empty
+    };
+    use dep::authwit::auth::{assert_current_call_valid_authwit, assert_current_call_valid_authwit_public, compute_authwit_nullifier};
+    use crate::types::nft_note::{NFTNote, NFTNoteHidingPoint};
+
+    global TRANSIENT_STORAGE_SLOT_PEDERSEN_INDEX = 3;
+
+    // TODO: Rename this to Transfer - calling this Transfer2 to avoid export conflict with the Transfer event in
+    // the Token contract
+    #[aztec(event)]
```

Comment:
Probably not that hard to fix. Just felt too low of a priority to bother with. Also I don't really emit the event anywhere yet. Would you say I should do that?

I am a bit bearish on Aztec events because they will be way more expensive than ethereum events so I would not be surprised if devs will end up not using them and instead will be analyzing calldata or smt.

But maybe Ethereum devs will save us, make DA super cheap and none of this will matter.

<img width="599" alt="image" src="https://github.com/user-attachments/assets/27825f2f-6ffe-4d8f-a528-2a826eb9e6ab">

Reply:
Good point. Let's create an issue about this for now (maybe export as NFTTransfer?).

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/8467